org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -575,6 +598,27 @@ public Preferences setImportOrder(List<String> importOrder) {
 		return this;
 	}
 
+	public Preferences setMaxBuildCount(int maxConcurrentBuilds) {
+		this.parallelBuildsCount = maxConcurrentBuilds;
+
+		String stringValue = maxConcurrentBuilds != 1 ? Boolean.TRUE.toString() : Boolean.FALSE.toString();
+
+		IEclipsePreferences pref = InstanceScope.INSTANCE.getNode(IMavenConstants.PLUGIN_ID);
+		pref.put(MavenPreferenceConstants.P_BUILDER_USE_NULL_SCHEDULING_RULE, stringValue);
+		pref = InstanceScope.INSTANCE.getNode(JavaCore.PLUGIN_ID);
+		pref.put(JavaBuilder.PREF_NULL_SCHEDULING_RULE, stringValue);
+		IWorkspace workspace = ResourcesPlugin.getWorkspace();
+		IWorkspaceDescription description = workspace.getDescription();
+
+		description.setMaxConcurrentBuilds(maxConcurrentBuilds);
+		try {
+			workspace.setDescription(description);
+		} catch (CoreException e) {
+			JavaLanguageServerPlugin.logException("Problems setting maxConcurrentBuilds from workspace.", e);
+		}
```

Comment:
All that preference management belongs to the PreferenceManager. Please move the code over there